lib/ansible/modules/system/aix_lvol.py

```diff
@@ -0,0 +1,352 @@
+        if match is not None:
+            pp_size = int(match.group(1))
+            continue
+        match = re.search("INTER-POLICY:\s+(\w+)", line)
+        if match is not None:
+            policy = match.group(1)
+            continue
+
+    if not name:
+        return None
+
+    size = lps * pp_size
+
+    return {'name': name, 'vg': vg, 'size': size, 'policy': policy}
+
+def parse_vg(data):
+
+    for line in data.splitlines():
+
+        match = re.search("VOLUME GROUP:\s+(\w+)", line)
+        if match is not None:
+            name = match.group(1)
+            continue
+
+        match = re.search("TOTAL PP.*\((\d+)", line)
+        if match is not None:
+            size = int(match.group(1))
+            continue
+
+        match = re.search("PP SIZE:\s+(\d+)", line)
+        if match is not None:
+            pp_size = int(match.group(1))
+            continue
+
+        match = re.search("FREE PP.*\((\d+)", line)
+        if match is not None:
+            free = int(match.group(1))
+            continue
+
+    return {'name': name, 'size': size, 'free': free, 'pp_size': pp_size}
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            vg=dict(required=True),
+            lv=dict(required=True),
+            lv_type=dict(default='jfs2'),
+            size=dict(type='str'),
+            opts=dict(default=''),
+            copies=dict(default='1'),
+            state=dict(choices=["absent", "present"], default='present'),
+            policy=dict(choices=["maximum", "minimum"], default='maximum'),
+            pvs=dict(type='list',default=list())
+        ),
+        supports_check_mode=True,
+    )
+
+    vg = module.params['vg']
+    lv = module.params['lv']
+    lv_type = module.params['lv_type']
+    size = module.params['size']
+    opts = module.params['opts']
+    copies = module.params['copies']
+    policy = module.params['policy']
+    state = module.params['state']
+    pvs = module.params['pvs']
+
+    pv_list = ' '.join(pvs)
+
+    if policy == "maximum":
+        lv_policy="x"
+    else:
+        lv_policy="m"
+
+    # Add echo command when running in check-mode
+    if module.check_mode:
+        test_opt = 'echo '
+    else:
+        test_opt = ''
+
+    # Get information on volume group requested
+    lsvg_cmd = module.get_bin_path("lsvg", required=True)
+    rc, vg_info, err = module.run_command("%s %s" % (lsvg_cmd, vg))
+
+    if rc != 0:
+        if state == 'absent':
+            module.exit_json(changed=False, msg="Volume group %s does not exist." % vg)
+        else:
+            module.fail_json(msg="Volume group %s does not exist." % vg, rc=rc, err=err)
+
+    this_vg = parse_vg(vg_info)
+
+
+    if size is not None:
+        # Calculate pp size and round it up based on pp size.
+        lv_size = round_ppsize(convert_size(size), base=this_vg['pp_size'])
+
+    # Get information on logical volume requested
+    lslv_cmd = module.get_bin_path("lslv", required=True)
```

Comment:
It would IMHO make more sense to move that near the other command, so we can verify both command at the same time.